File: abc/095/c.rb
a, b, c, x, y = gets.split.map(&:to_i)

if c < 0.5 * (a + b)
  if x < y
    puts c * 2 * x + [b, 2 * c].min * (y - x)
  else
    puts c * 2 * y + [a, 2 * c].min * (x - y)
  end
else
  puts a * x + b * y
end

File: abc/095/c.py
a, b, c, x, y = map(int, input().split())

min_price = 2 * 5000 * 100000
for nab in range(0, 2 * max(x, y) + 1, 2):
    na = max(0, x - nab // 2)
    nb = max(0, y - nab // 2)
    min_price = min(min_price, nab * c + na * a + nb * b)

print(min_price)
